### PR TITLE
tests/driver_hd44780: allow use on non-Arduino boards

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description
Currently, `tests/driver_hd44780` depends on the Arduino module and hard-codes an Arduino default configuration.

This PR moves the default configuration to the driver like it is done in other such drivers and allows it to be overwritten by the board configuration.

When hooking up the 16x2 display on the MCB2300 board, the display would not light up.
I had to add a short delay (1µs was enough) after each byte transferred, then the display would update.

Curiously, increasing `HD44780_PULSE_WAIT_LONG` did not help.

### Testing procedure

Run `tests/driver_hd44780` and check that the display still prints the test messages.

### Issues/PRs references
none

### CODE REVIEW

File: drivers/hd44780/include/hd44780_params.h [hunk 1]
```diff
@@ -38,12 +40,31 @@ extern "C"
                HD44780_RW_OFF, HD44780_RW_OFF, HD44780_RW_OFF, HD44780_RW_OFF} \
 }
 
+#ifndef HD44780_PARAMS
+#define HD44780_PARAMS HD44780_PARAMS_ARDUINO
+#endif
+#endif
```
Comment (leandrolanzieri) [inline drivers/hd44780/include/hd44780_params.h]: Please add comments on the `#endif`

File: drivers/hd44780/include/hd44780_params.h [hunk 2]
```diff
@@ -38,12 +40,31 @@ extern "C"
...
+
+#ifndef HD44780_PARAMS
+#define HD44780_PARAMS {            \
```
Comment (leandrolanzieri) [inline drivers/hd44780/include/hd44780_params.h]: Is this setup convenient for a particular board? If so, could you spare a comment mentioning it just for reference?

Comment (benpicco) [reply drivers/hd44780/include/hd44780_params.h]: No, it's just ascending numbers.
I wanted to make sure that overwriting the config with the board config works.

File: drivers/hd44780/include/hd44780_params.h (hunk 1 above)
Comment (benpicco) [reply drivers/hd44780/include/hd44780_params.h]: Added.

**Comment by leandrolanzieri** [comment]
@benpicco or @smlng do you know if the arduino pin configuration refers to some particular shield?

**Comment by smlng** [comment]
the display is available in 2 variants: one with a generic GPIO interface and another with I2C. The driver is written for the GPIO interface (more wires) and PIN mappings are for standard Arduino header - no shield.

**Comment by aabadie** [comment]
I'm sorry this is a duplicate of #12487 (but adds the missing delay). I think #12487 goes a little further in cleaning up the driver.

**Comment by benpicco** [comment]
duplicate of #12487

**Pull request closed without merging**